projects/packages/backup/tests/php/test-rest-controller.php

```diff
@@ -166,6 +172,219 @@ public function test_delete_backup_helper_script_success() {
+			)
+		);
+		$response      = $this->server->dispatch( $request );
+		$response_data = $response->get_data();
+		$this->assertEquals( 400, $response->get_status() );
+		$this->assertEquals( 'Invalid parameter(s): object_id', $response_data['message'] );
+		$this->assertEquals( 'object_id is not of type integer.', $response_data['data']['params']['object_id'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/database-object/backup` endpoint with admin user.
+	 */
+	public function test_backup_database_object_unauthorized() {
+		wp_set_current_user( $this->admin_id );
+
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/database-object/backup' );
+		$request->set_query_params(
+			array(
+				'object_id'   => 123,
+				'object_type' => 'woocommerce_attribute',
+			)
+		);
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 403, $response->get_status() );
+		$this->assertEquals( 'You are not allowed to perform this action.', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/database-object/backup` endpoint on success.
+	 */
+	public function test_backup_database_object_success() {
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/database-object/backup' );
+		$request->set_query_params(
+			array(
+				'object_id'   => 123,
+				'object_type' => 'woocommerce_attribute',
+			)
+		);
+		$response = $this->server->dispatch( $request );
+
+		$response = $this->dispatch_request_signed_with_blog_token( $request );
+
+		$this->assertEquals( 404, $response->get_status() ); // success in this context.
+
+		$this->assertEquals( 'Object not found', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/options/backup` endpoint with invalid params.
+	 */
+	public function test_backup_options_invalid_params() {
+		$request  = new WP_REST_Request( 'GET', '/jetpack/v4/options/backup' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 400, $response->get_status() );
+		$this->assertEquals( 'Missing parameter(s): name', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/options/backup` endpoint with admin user.
+	 */
+	public function test_backup_options_unauthorized() {
+		wp_set_current_user( $this->admin_id );
+
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/options/backup' );
+		$request->set_query_params(
+			array(
+				'name' => 'home',
+			)
+		);
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 403, $response->get_status() );
+		$this->assertEquals( 'You are not allowed to perform this action.', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/options/backup` endpoint on success.
+	 */
+	public function test_backup_options_success() {
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/options/backup' );
+		$request->set_query_params(
+			array(
+				'name' => 'home',
+			)
+		);
+		$response = $this->server->dispatch( $request );
```

Comment:
I think this `server->dispatch` call is unnecessary, since we call `dispatch_request_signed_with_blog_token` below.